libz-rs-sys/src/gz.rs

```diff
@@ -172,7 +175,18 @@ unsafe fn gzopen_help(path: *const c_char, fd: c_int, mode: *const c_char) -> gz
     if fd > -1 {
         state.fd = fd;
     } else {
-        let mut oflag: c_int = 0;
+        let mut oflag = 0;
+
+        #[cfg(target_os = "linux")]
+        {
+            oflag |= libc::O_LARGEFILE;
+        }
+
+        #[cfg(target_os = "linux")]
+        if cloexec {
+            oflag |= libc::O_CLOEXEC;
+        }
+
         if state.mode == GzMode::GZ_READ {
             oflag |= O_RDONLY;
         } else {
```

Comment:
I can't comment on it on the line, but I think the block for `if state.mode == GzMode::GZ_APPEND` should be moved out of the else branch, and should also run if the user did provide a file descriptor. At least, that is how I read the C code.

Reply:
there are some missing comments there as well